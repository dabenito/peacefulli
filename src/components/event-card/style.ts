import { StyleSheet } from 'react-native';
import { Color } from '../../constants/color-constants';

export const styles = StyleSheet.create({
    container: {
        flex: 1,
        alignItems: 'center',
        justifyContent: 'center',
    },
    image: {
        width: 200,
        height: 100,
        borderRadius: 10,
        marginTop: 30,
        marginBottom: 5,
    },
    textPrimary: {
        color: Color.black,
        fontFamily: 'lucida_grande_bold',
        fontSize: 15,
        marginTop: 5,
        marginLeft: 10,
        marginRight: 10,
    },
    textSecondary: {
        color: Color.black,
        fontFamily: 'lucida_grande_regular',
        fontSize: 13,
        marginTop: 5,
        marginLeft: 30,
        marginRight: 30,
    },
});
