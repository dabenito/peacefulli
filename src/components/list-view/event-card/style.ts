import { StyleSheet } from "react-native";
import { Color } from "../../../constants/color-constants";

export const styles = StyleSheet.create({
  cardContainer: {
    display: "flex",
    flexDirection: "row",
    marginTop: 16,
    marginLeft: 16,
    marginRight: 16,
  },
  imageContainer: {
    width: "33%",
    marginRight: 6,
  },
  textContainer: {
    width: "58%",
  },
  favoriteContainer: {
    width: "7%",
  },
  locationContainer: {
    flexDirection: "row",
    marginTop: 4,
  },
  image: {
    width: 120,
    height: 120,
    marginRight: 8,
  },
  dateText: {
    color: Color.purple,
    fontFamily: "pt_sans_regular",
    fontSize: 14,
  },
  titleText: {
    color: Color.black,
    fontFamily: "pt_sans_bold",
    fontSize: 16,
    marginTop: 4,
  },
  descriptionText: {
    color: Color.dark_grey,
    fontFamily: "pt_sans_regular",
    flexWrap: "wrap",
    fontSize: 11,
    marginTop: 4,
  },
  locationText: {
    color: Color.black,
    fontFamily: "pt_sans_italic",
    fontSize: 10,
    marginLeft: 4,
  },
  divider: {
    backgroundColor: Color.light_grey,
    marginTop: 16,
    height: 2,
    opacity: 0.5,
  },
  locationPin: {
    height: 10,
  },
  favoriteIcon: {
    height: 24,
  },
});
