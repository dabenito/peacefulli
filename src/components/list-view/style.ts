import { StyleSheet } from 'react-native';
import { Color } from '../../constants/color-constants';

export const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: Color.white,
        alignItems: 'center',
        justifyContent: 'center',
    },
    listContainer: {
        flex: 4,
        backgroundColor: Color.white,
        alignItems: 'center',
        justifyContent: 'center',
        marginTop: -75,
    },
});
