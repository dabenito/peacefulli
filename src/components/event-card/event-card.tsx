
import React, { PureComponent } from 'react';
import {
    Text,
    Image,
    ImageSourcePropType,
    TouchableOpacity
} from "react-native";
import { styles } from './style';
import { customFonts } from '../../constants/font-constants';
import * as Font from 'expo-font';

export interface EventCardProps {
    imageSource: ImageSourcePropType
    date: String
    location: String
    description: String
}

export interface EventCardState {
    fontLoaded: boolean
}

export class EventCard extends PureComponent<EventCardProps, EventCardState>{
    constructor(props: EventCardProps) {
        super(props);
        this.state = { fontLoaded: false };
    }

    onPress() {
        console.log("The event was pressed");
    }

    async _loadFontsAsync() {
        await Font.loadAsync(customFonts);
        this.setState({ fontLoaded: true });
    }

    componentDidMount() {
        this._loadFontsAsync();
    }

    render() {
        if (this.state && this.state.fontLoaded) {
            return (
                <TouchableOpacity onPress={this.onPress} style={styles.container}>
                    <Image style={styles.image} source={{ uri: 'https://i.stack.imgur.com/khVpE.png' }} />
                    <Text style={styles.textPrimary}>{this.props.date + " - " + this.props.location}</Text>
                    <Text style={styles.textSecondary}>{this.props.description}</Text>
                </TouchableOpacity>
            );
        } else {
            return null;
        }
    }
}