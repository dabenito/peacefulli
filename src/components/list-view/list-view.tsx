import React, { PureComponent } from "react";
import { View, Text } from "react-native";
import { styles } from "./style";

export interface ListViewProps {}

export interface ListViewState {}

export class ListView extends PureComponent<ListViewProps, ListViewState> {
  constructor(props: ListViewProps) {
    super(props);
  }

  render() {
    return (
      <View style={styles.container}>
        <Text>TODO: Create List View!</Text>
      </View>
    );
  }
}
