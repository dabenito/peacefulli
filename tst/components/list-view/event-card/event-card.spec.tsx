import React from "react";
import { render, cleanup } from "@testing-library/react-native";
import { EventCard } from "../../../../src/components/list-view/event-card";

describe("EventCard Tests", () => {
  afterEach(cleanup);

  it("event card with favorite enabled", async () => {
    const eventCard = render(
      <EventCard
        id="randomId"
        date="date"
        title="title"
        description="description"
        location="location"
        imageSource="url"
        isFavorite={true}
      />
    );
    expect(eventCard).toMatchSnapshot();
  });

  it("event card with favorite disabled", async () => {
    const eventCard = render(
      <EventCard
        id="randomId"
        date="date"
        title="title"
        description="description"
        location="location"
        imageSource="url"
        isFavorite={false}
      />
    );
    expect(eventCard).toMatchSnapshot();
  });
});
