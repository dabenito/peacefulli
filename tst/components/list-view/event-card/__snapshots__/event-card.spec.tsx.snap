// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventCard Tests event card with favorite disabled 1`] = `
<View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "marginLeft": 16,
        "marginRight": 16,
        "marginTop": 16,
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "marginRight": 6,
          "width": "33%",
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "url",
          }
        }
        style={
          Object {
            "height": 120,
            "marginRight": 8,
            "width": 120,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "width": "58%",
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Object {
            "color": "#6900FF",
            "fontFamily": "pt_sans_regular",
            "fontSize": 14,
          }
        }
      >
        date
      </Text>
      <Text
        numberOfLines={1}
        style={
          Object {
            "color": "#000000",
            "fontFamily": "pt_sans_bold",
            "fontSize": 16,
            "marginTop": 4,
          }
        }
      >
        title
      </Text>
      <Text
        ellipsizeMode="tail"
        numberOfLines={3}
        style={
          Object {
            "color": "#6C6C6C",
            "flexWrap": "wrap",
            "fontFamily": "pt_sans_regular",
            "fontSize": 11,
            "marginTop": 4,
          }
        }
      >
        description
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
            "marginTop": 4,
          }
        }
      >
        <Text />
        <Text
          numberOfLines={1}
          onPress={[Function]}
          style={
            Object {
              "color": "#000000",
              "fontFamily": "pt_sans_italic",
              "fontSize": 10,
              "marginLeft": 4,
            }
          }
        >
          location
        </Text>
      </View>
    </View>
    <Text />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#CCCCCC",
        "height": 2,
        "marginTop": 16,
        "opacity": 0.5,
      }
    }
  />
</View>
`;

exports[`EventCard Tests event card with favorite enabled 1`] = `
<View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "marginLeft": 16,
        "marginRight": 16,
        "marginTop": 16,
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "marginRight": 6,
          "width": "33%",
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "url",
          }
        }
        style={
          Object {
            "height": 120,
            "marginRight": 8,
            "width": 120,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "width": "58%",
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Object {
            "color": "#6900FF",
            "fontFamily": "pt_sans_regular",
            "fontSize": 14,
          }
        }
      >
        date
      </Text>
      <Text
        numberOfLines={1}
        style={
          Object {
            "color": "#000000",
            "fontFamily": "pt_sans_bold",
            "fontSize": 16,
            "marginTop": 4,
          }
        }
      >
        title
      </Text>
      <Text
        ellipsizeMode="tail"
        numberOfLines={3}
        style={
          Object {
            "color": "#6C6C6C",
            "flexWrap": "wrap",
            "fontFamily": "pt_sans_regular",
            "fontSize": 11,
            "marginTop": 4,
          }
        }
      >
        description
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
            "marginTop": 4,
          }
        }
      >
        <Text />
        <Text
          numberOfLines={1}
          onPress={[Function]}
          style={
            Object {
              "color": "#000000",
              "fontFamily": "pt_sans_italic",
              "fontSize": 10,
              "marginLeft": 4,
            }
          }
        >
          location
        </Text>
      </View>
    </View>
    <Text />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#CCCCCC",
        "height": 2,
        "marginTop": 16,
        "opacity": 0.5,
      }
    }
  />
</View>
`;
